Corgibytes.Freshli.Cli/Functionality/Git/CloneGitRepositoryActivity.cs

```diff
@@ -1,25 +1,48 @@
+using System;
 using Corgibytes.Freshli.Cli.Exceptions;
 using Corgibytes.Freshli.Cli.Functionality.Engine;
+using Microsoft.Extensions.DependencyInjection;
 using Newtonsoft.Json;
 
 namespace Corgibytes.Freshli.Cli.Functionality.Git;
 
 public class CloneGitRepositoryActivity : IApplicationActivity
 {
-    [JsonProperty] private readonly string? _branch;
+    [JsonProperty] private readonly ICachedGitSourceRepository _gitSourceRepository;
+
     [JsonProperty] private readonly string _cacheDir;
     [JsonProperty] private readonly string _gitPath;
-    [JsonProperty] private readonly ICachedGitSourceRepository _gitSourceRepository;
+    [JsonProperty] private readonly Guid _analysisId;
+
     [JsonProperty] private readonly string _repoUrl;
+    [JsonProperty] private readonly string? _branch;
 
+    [JsonConstructor]
     public CloneGitRepositoryActivity(ICachedGitSourceRepository gitSourceRepository,
-        string repoUrl, string? branch, string cacheDir, string gitPath)
+        string repoUrl, string? branch, string cacheDir, string gitPath, Guid analysisId = new())
     {
         _gitSourceRepository = gitSourceRepository;
         _repoUrl = repoUrl;
         _branch = branch;
         _cacheDir = cacheDir;
         _gitPath = gitPath;
+        _analysisId = analysisId;
+    }
+
+    public CloneGitRepositoryActivity(IServiceProvider serviceProvider, Guid analysisId, string cacheDir, string gitPath)
+    {
+        _cacheDir = cacheDir;
+        _gitPath = gitPath;
+        _analysisId = analysisId;
+
+        _gitSourceRepository = serviceProvider.GetRequiredService<ICachedGitSourceRepository>();
```

Comment:
I've been contemplating whether or not to take this approach with some of the activities. I like seeing it here so we can compare and contrast it against the other approaches and decide on the one that we like most.